test/integration-tests/Uniswap.test.ts

```diff
@@ -587,6 +589,64 @@ describe('Uniswap V2 and V3 Tests:', () => {
         expect(ethBalanceBefore.sub(ethBalanceAfter)).to.eq(wethTraded.add(gasSpent))
       })
     })
+
+    describe('STETH', () => {
+      const STETH_DUST = 1
+
+      it('completes a V3 exactIn with STETH --> WETH', async () => {
+        const { stethContract } = await setupStethContext()
+
+        const amountInSTETH: number = expandTo18DecimalsBN(0.1)
+        const amountInWSTETH: number = await stethContract.getSharesByPooledEth(amountInSTETH.sub(STETH_DUST))
+        const amountOutMin: number = expandTo6DecimalsBN(0)
+
+        planner.addCommand(CommandType.PERMIT2_TRANSFER_FROM, [STETH.address, ADDRESS_THIS, amountInSTETH])
+        planner.addCommand(CommandType.WRAP_STETH, [ADDRESS_THIS, CONTRACT_BALANCE])
+        addV3ExactInTrades(
+          planner,
+          1,
+          amountOutMin,
+          MSG_SENDER,
+          [WSTETH.address, WETH.address],
+          SOURCE_ROUTER,
+          amountInWSTETH
+        )
+
+        const stethBalanceBefore = await stethContract.balanceOf(bob.address)
+        const { wethBalanceBefore, wethBalanceAfter } = await executeRouter(planner)
+        const stethBalanceAfter = await stethContract.balanceOf(bob.address)
+
+        expect(stethBalanceBefore.sub(stethBalanceAfter)).to.eq(amountInSTETH.sub(STETH_DUST))
```

Comment:
i'm a little confused on what this balance check is checking and what STETH_DUST represents, so when we wrap into wSETH, there's some dust that is lost that we have to account for here?

Reply:
yah, wheneverrrrr STETH is tranferred there's always 1 wei dust 😓

Reply:
whatt that's crazy